decidim-accountability/lib/decidim/accountability/query_extensions.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Accountability
+    # This module's job is to extend the API with custom fields related to
+    # decidim-proposals.
+    module QueryExtensions
+      # Public: Extends a type with `decidim-proposals`'s fields.
+      #
+      # type - A GraphQL::BaseType to extend.
+      #
+      # Returns nothing.
+      def self.define(type)
+        type.field :resultsMetric, Accountability::ResultsMetricType, "Decidim's ResultMetric data." do
+          resolve lambda { |_obj, _args, ctx|
+            ctx[:current_organization]
```

Comment:
All metrics are scope inside an organization

Reply:
But what's this line doing? Isn't this returning the organization? 😕

Reply:
Yes, in the context of request

Reply:
OK, I see the problem. If you don't put a resolver here, then GraphQL tries to call `object.resultsMetric`, where `object` is whatever the query is scoped to. Since it's not scoped to anything (it's a first-level query), it calls it on `nil`, which fails, so you need to make this resolve to something meaningful. I tried making it resolve to `nil`, but it fails as per how GraphQL works.

The `resolve` does not pass the organization to subsequent queries. It "resolves" how to calculate this field. That's why in your `count` fields you make it calculate that count. Whatever the field is resolved to, it's passed as the `obj` in the subsequent queries.

That's why returning the organization here feels very weird. Resolving `resultsMetric` as an organization is not accurate. 

I see all metrics have the same interface: they have a `count` and a `metric` field. One way to make this part of the code clearer is making this `resultsMetric` field resolve to an object that responds to `count` and `metric`, and knows how to calculate these values.

```ruby
type.field :resultsMetric, Accountability::ResultsMetricType, "Decidim's ResultMetric data." do
  resolve lambda { |_obj, _args, ctx| Decidim::Accountability::API::ResultsMetricResolver.new(ctx[:current_organization]) }
end
```

Internally, this `ResultsMetricResolver` defines the `count` and `metric` methods, which know how to resolve the code that currently lies in the `ResultsMetricTypeHelper`